### PR TITLE
Run insert queries on the main database

### PR DESCRIPTION
**Pull request by wyrzyk**

### CODE REVIEW

**Review by dwatl — DISMISSED** [review]

File: src/main/java/com/atlassian/db/replica/internal/SqlQuery.java [hunk 1]
```diff
@@ -13,14 +13,18 @@ public SqlQuery(String sql) {
...
+        return sqlFunction.isFunctionCall(sql) || isUpdate() || isDelete() || isInsert();
     }
 
     public boolean isSelectForUpdate() {
         final String trimmedQuery = trimForSelectForUpdateCheck();
         return trimmedQuery.contains("for update") || trimmedQuery.contains("FOR UPDATE");
     }
 
+    public boolean isInsert() {
+        return sql.startsWith("insert") || sql.startsWith("INSERT");
```
Comment (afaruga-atlassian) [inline src/main/java/com/atlassian/db/replica/internal/SqlQuery.java:25]: `sql.contains("insert")`

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java [hunk 2]
```diff
@@ -54,6 +54,23 @@
...
 
+    @Test
+    public void shouldUseMainConnectionForInserts() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            permanentConsistency().build()
+        ).build();
+
+        connection.prepareStatement("insert into \"public\".\"fvv\" (\"cd\", \"iid\", \"did\", \"s\")\n" +
```
Comment (afaruga-atlassian) [inline src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:65]: Did that happens on production and results with failures from Replica ?
Asking cause I haven't see it yet.

File: src/main/java/com/atlassian/db/replica/internal/SqlQuery.java (hunk 1 above)
Comment (wyrzyk) [reply src/main/java/com/atlassian/db/replica/internal/SqlQuery.java:25]: Most db-replicas decisions are heuristics focused on performance over accuracy. We update the rules when we see it doesn't work on prod. String#contains is O(nm) (for the Java implementations I've seen. It's possible to implement it better O(n), but it's still costly), and startsWith is O(1) and passes the case from prod.

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java (hunk 2 above)
Comment (wyrzyk) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:65]: Yes. See the internal stand-up channel.

File: src/main/java/com/atlassian/db/replica/internal/SqlQuery.java (hunk 1 above)
Comment (afaruga-atlassian) [reply src/main/java/com/atlassian/db/replica/internal/SqlQuery.java:25]: `Most db-replicas decisions are heuristics focused on performance over accuracy. `

👍  Ok, lets follow this

File: src/test/java/com/atlassian/db/replica/api/TestDualConnection.java (hunk 2 above)
Comment (afaruga-atlassian) [reply src/test/java/com/atlassian/db/replica/api/TestDualConnection.java:65]: `"ERROR: cannot execute INSERT in a read-only transaction"`

I see

**Review by afaruga-atlassian — APPROVED** [review]

**Pull request merged**